### ISSUE TITLE
Decide when published messages with message id should be removed from state

### ISSUE THREAD

**Issue by Zelldon** [zeebe-io/zeebe#1323]
With #1291 with introduced a new state controller for storing messages and message subscriptions, since this is now a map instead of a list we overwrite messages with the same name and correlation key. 

But this is only true for the default column family. For `timeToLive` and `messageId`, where the key is composite, with the time or the message id we get orphan entries. We should decide, when we want to clean them up. 

Possible way are on put:

```java
   ...
    final Message oldMessage =
        getMessage(keyBuffer, KEY_OFFSET, message.getKeyLength() - KEY_OFFSET);
    if (oldMessage != null) {
      remove(oldMessage);
    }
  ...
```

Or on remove call and remove all related entries and old entries (need to iterate over the column families).

This would be OK for the `timeToLive` column family, but what is with the message id.
If we remove the message with this id, then the existence checks will returns false and we are not able to reject the message with the same id anymore.

**Comment by npepinpe** [comment]
Regarding your last point, so when we remove a message, we need to still remember that it was once published? Is that a requirement? It makes sense to me that when you receive a delete message command, you would remove all trace of it, and be able to republish it (so no need to remember the ID for the existence check).

**Comment by Zelldon** [comment]
> Regarding your last point, so when we remove a message, we need to still remember that it was once published? Is that a requirement?

This is the point where we are not sure. With the message ID wanted to guarantee that a message is not published twice, but how long? For ever? :sweat_smile:

**Comment by saig0** [comment]
No, only so long as the message exist. After that you can publish it again.

**Comment by Zelldon** [comment]
Aah nice.Thanks @saig0 that makes it easier, then I have to impl that.

**Comment by saig0** [comment]
The assumption that messages with the same name and correlation key can be overridden is not true. All messages have to be stored under a unique key. Since the messages have a unique key, we can simply delete the message and all references by the key when the TTL is reached.

**Comment by saig0** [comment]
The state and the deletion will be fixed by #1405.
